File: demo/java-demo/src/main/java/com/alibaba/demo/basic/DemoPrivateAccess.java
package com.alibaba.demo.basic;

import java.util.List;

/**
 * 演示私有成员访问功能
 * Demonstrate private member access functionality
 */
public class DemoPrivateAccess {

    /**
     * a private static field
     */
    private static int staticCount;
    /**
     * a private member field
     */
    private int count;
    /**
     * a constant field
     */
    public final Double pi = 3.14;

    /**
     * private static method without arguments
     */
    private static String privateStaticFunc() {
        return "static";
    }

    /**
     * private static method with arguments
     */
    private static String privateStaticFuncWithArgs(String str, int i) {
        return (str == null ? "null" : str) + " + " + i;
    }

    /**
     * private member method without arguments
     */
    private String privateFunc() {
        return "member";
    }

    /**
     * private member method with arguments
     */
    private String privateFuncWithArgs(List<String> list, String str, int i) {
        return list.stream().reduce((a, s) -> a + s).orElse("") + " + " + str + " + " + i;
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/demo/basic/DemoPrivateAccess.kt
package com.alibaba.demo.basic

/**
 * 演示私有成员访问功能
 * Demonstrate private member access functionality
 */
class DemoPrivateAccess {

    /**
     * a private member field
     */
    var count = 0

    /**
     * a constant field
     */
    val pi = 3.14

    /**
     * private member method
     */
    fun privateFunc(list: List<String>, str: String, i: Int): String {
        return list.reduce { a: String, s: String -> a + s } + " + " + "$str + $i"
    }

    companion object {

        /**
         * a private static field
         */
        var staticCount = 0

        /**
         * private static method
         */
        fun privateStaticFunc(str: String, i: Int): String {
            return "$str + $i"
        }

        /**
         * private jvm static method
         */
        @JvmStatic private fun privateJvmStaticFunc(list: List<String>, str: String, i: Int): String {
            return list.reduce { a: String, s: String -> a + s } + " * " + "$str * $i"
        }
    }

}
